File: python/market_order.py
"""
Market Order Example

Place a market order that executes immediately at best available price.
"""

from hyperliquid_sdk import HyperliquidSDK

sdk = HyperliquidSDK()

# Market buy by size (0.0001 BTC)
order = sdk.market_buy("BTC", size=0.0001)
print(f"Market buy: {order}")
print(f"  Status: {order.status}")
print(f"  OID: {order.oid}")

# Market buy by notional ($10 worth of ETH)
# order = sdk.market_buy("ETH", notional=10)

# Market sell
# order = sdk.market_sell("BTC", size=0.0001)

File: typescript/market_order.ts
#!/usr/bin/env npx ts-node
/**
 * Market Order Example
 *
 * Place a market order that executes immediately at best available price.
 */

import { HyperliquidSDK } from 'hyperliquid-sdk';

async function main() {
  const sdk = new HyperliquidSDK();

  // Market buy by notional ($11 worth of BTC - minimum is $10)
  const order = await sdk.marketBuy("BTC", { notional: 11 });
  console.log(`Market buy: ${order}`);
  console.log(`  Status: ${order.status}`);
  console.log(`  OID: ${order.oid}`);

  // Market buy by notional ($10 worth of ETH)
  // const order = await sdk.marketBuy("ETH", { notional: 10 });

  // Market sell
  // const order = await sdk.marketSell("BTC", { size: 0.0001 });
}

main().catch(console.error);
